pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -33,14 +33,16 @@ const (
 
 // Go doesn't like this as a constant
 var (
-	downloaderObjTypes = []string{types.AppImgObj, types.BaseOsObj,
-		types.CertObj}
+	debug              = false
+	debugOverride      bool                               // From command line arg
+	downloadGCTime     = time.Duration(600) * time.Second // Unless from GlobalConfig
+	downloadRetryTime  = time.Duration(600) * time.Second // Unless from GlobalConfig
+	downloaderObjTypes = []string{types.AppImgObj, types.BaseOsObj, types.CertObj}
+	Version            = "No version specified" // Set from Makefile
+	nilUUID            uuid.UUID
+	dHandler           = &downloadHandler{}
```

Comment:
Why don't you do the init/make here and avoid the handlersCheckOrInit call in each invocation?

Reply:
We could. A common pattern is to _not_ call `&something{}` but instead `newSomething()` which initializes. I will update that right away.

Reply:
Updated